bin/merge_featurecounts.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env/python
+
+import argparse
+import os
+import re
+from collections import defaultdict
+
+
+def do_the_thing(dest_dir,out_file,input_files):
+   table_dict=defaultdict(dict)
+   sample_names=[]
+   gene_list=[]
+   for input_file in input_files:
+       if not gene_list:
+           first_file=True
+       print"Reading from {}".format(input_file)
+       sample_name=os.path.basename(input_file)
+       sample_name= sample_name.replace("Aligned.sortedByCoord.out_gene.featureCounts.txt","")
+       sample_names.append(sample_name)
+       table_dict[sample_name]=dict()
+     
+       #Extract the gene list from the first file
+       if first_file:
+           with open(input_file, 'r') as f:
+               for line in f:
+                   if not line.startswith('E'):
+                       continue
+                   line_info=line.split('\t')
+                   gene=line_info[0]
+                   gene_list.append(gene)
+       #Read the rest of the files
+       with open(input_file, 'r') as f:
+           for line in f:
+               if not line.startswith('E'):
+                   continue
+               line_info=line.split('\t')
+               gene=line_info[0]
+               gene_count=line_info[-1]
+               gene_count=gene_count.rstrip()
+               if gene_count.startswith('E'):
```

Comment:
What error did this come from? Might be nice to have a more generic version that asserts that we found a number?

```python
try:
    gene_count = int(line_info[-1].rstrip())
    table_dict[sample_name][gene] = gene_count
except TypeError:
    print "Detected discrepancy in {}  line {}".format(input_file, line)
```